public/js/index.js

```diff
@@ -11,6 +11,13 @@ console.info(`rq_ui config format version ${RQ_PARAMS.CONFIG_FORMAT_VERSION}`)
 
 const keyControl = new KeyControl('#keyControl')
 
+/**
+ * A simple helper function that expands on jQuery to allow the WIDGET_NAMESPACE
+ */
```

Comment:
This is a useful function. Please add the @returns javadocs attribute to describe what the function returns.

My understanding of this change is that it changes the name of the jQuery widget data object from 'widget' to RQ_PARAMS.WIDGET_NAMESPACE. If that's correct, I'm not convinced that's optimal, because it will imply the data belongs to anything and everything in the namespace instead of to just this one widget. With that said, I do agree that using the name 'widget' is sub-optimal, because it doesn't distinguish between the general purpose jQuery concept of a widget and this one specific RoboQuest widget.
See my comment further down about the new value of RQ_PARAMS.WIDGET_NAMESPACE.

Reply:
If this is the right approach, I'll definitely add the @returns, thank you!

I think your understanding is correct, but just to be sure, the purpose of this function is simply to return the same value that `.data('widget')` would've, the benifit being that we don't need to repeat `RQ_PARAMS.WIDGET_NAMESPACE` over and over.

Did we want to just rename the string 'widget' to RQ_PARAMS.WIDGET_NAMESPACE? That's what I assumed, but now I'm not too sure.

Reply:
Yes. I believe using the namespace 'rq' to refer to the data() object is more clear than using 'widget'. That will allow us to assumed that anywhere we see the string 'widget', we're referring to a general purpose jQuery widget. When we're referring specifically to a RoboQuest widget (which, of course, was created by the jQuery Widget Factory, we'll first put them into the 'rq' namespace. For variables which refer to RoboQuest widgets, we can use rqWidget or rqWidgetObj as appropriate.

The overall goal I'm working toward is: I read a variable name and there's minimal uncertainty about what it contains AS WELL AS when I assign a name to or for something, I then use that name exclusively and consistently throughout the entire code base for that one concept or usage.